src/Cake.Core/IO/IDirectory.cs

```diff
@@ -14,6 +14,12 @@ public interface IDirectory
         DirectoryPath Path { get; }
 
         /// <summary>
+        /// Gets the name of the directory.
+        /// </summary>
+        /// <value>The directory's name.</value>
+        string Name { get; }
```

Comment:
Duly noted, and change will be made. Though I'd put in my two cents that Name is something that'd be good for both File and Directory to have, if only for consistency with framework patterns ^_^.